kspmanager.py

```
@@ -158,41 +206,91 @@ def get_craft_path(self, name):
+                        from PyQt4 import QtCore
+                        self.emitter.emit(QtCore.SIGNAL('building(QString, int, int)'), "Building", list(assets_set).index(asset), len(assets_set))
+                    print('Converting textures for {}'.format(asset))
+                    craft_assets.update(self.get_with_converted_textures(self.parts[asset]))
+        return craft_assets
+
+    def build_zip(self, craft_name, craft_file, craft_assets):
         output = tempfile.gettempdir()
         archive = os.path.join(output, craft_name + '.zip')
+        self.temp_files.add(archive)
         zip = zipfile.ZipFile(archive, 'w')
         zip.write(craft_file, os.path.basename(craft_file))
-        map(zip.write, set(reduce(lambda x, y: x + y,
-                map(list_folder_files, folders))))
+        # When textures are converted, PNGs are put in the archive from the temp path but
+        # with the MBM (game) path to keep their relative path with the model, inside the archive
+        print('Building the .zip')
+        for f in craft_assets:
+            if isinstance(f, tuple):
+                zip.write(f[0], f[1])
+            else:
+                zip.write(f)
         zip.close()
+
         return archive
 
 
+    def clear_tmp_files(self):
+        ''' Clear temp files '''
+        print('Cleaning temp files...')
+        for d in self.temp_files:
+            if tempfile.gettempdir() in d:
+                if os.path.isdir(d):
+                    for f in os.listdir(d):
+                        try:
+                            os.remove(os.path.join(d,f))
+                        except:
+                            pass
+                    try:
+                        os.rmdir(d)
+                    except:
+                        pass
+                else:
+                    try:
+                        os.remove(d)
+                    except:
+                        pass
+        print('Temp files cleaned')
```

a lot of `if`s, a lot of catch all exceptions. Is this really necessary? Tempfiles are meant to be deleted upon computer restart